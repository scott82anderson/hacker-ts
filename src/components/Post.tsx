import React from 'react'
import { Link } from 'react-router-dom'
import { Post as IPost } from '../utils/api'
import { formatDate } from '../utils/helpers'

function Post(post: IPost): JSX.Element {
    return <li className="py-6 border-b border-gray-100">
        <a target="_blank" rel="noreferrer" className="font-semibold text-blue-900 dark:text-yellow-500 underline hover:no-underline" href={post.url}>{post.title}</a>
        <div className="meta-info-light"><span>by <Link className="text-blue-900 dark:text-yellow-500 underline hover:no-underline" to={`/user?id=${post.by}`}>{post.by}</Link></span>
            <span className="font-semibold"> on {formatDate(post.time)}</span><span> with <Link className="text-blue-900 dark:text-yellow-500 underline hover:no-underline" to={`/post?id=${post.id}`}>{post.descendants}</Link> comments</span>
            </div>
        </li>
  
}

export default Post
