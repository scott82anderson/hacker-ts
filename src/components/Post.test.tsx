import React from 'react';
import { render, screen } from '@testing-library/react';
import Post from './Post';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

const testPost = {
  id: 1,
  by: "author",
  time: 946720800,
  text: "The post body",
  title: "Post title",
  url: "https://www.google.com",
  descendants: 10,
  kids: [],
  type: "post"
}

test('render a Post', () => {
  render(<Router><Post {...testPost} /></Router>);
  const titleElement = screen.getByText(/Post title/i);
  expect(titleElement).toBeInTheDocument();
});


test('Post snapshot', () => {
  const tree = renderer
    .create(<Router><Post {...testPost} /></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
