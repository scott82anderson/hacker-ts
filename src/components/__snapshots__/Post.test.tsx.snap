// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post snapshot 1`] = `
<li
  className="py-6 border-b border-gray-100"
>
  <a
    className="font-semibold text-blue-900 dark:text-yellow-500 underline hover:no-underline"
    href="https://www.google.com"
    rel="noreferrer"
    target="_blank"
  >
    Post title
  </a>
  <div
    className="meta-info-light"
  >
    <span>
      by 
      <a
        className="text-blue-900 dark:text-yellow-500 underline hover:no-underline"
        href="/user?id=author"
        onClick={[Function]}
      >
        author
      </a>
    </span>
    <span
      className="font-semibold"
    >
       on 
      01/01/2000, 9:00 pm
    </span>
    <span>
       with 
      <a
        className="text-blue-900 dark:text-yellow-500 underline hover:no-underline"
        href="/post?id=1"
        onClick={[Function]}
      >
        10
      </a>
       comments
    </span>
  </div>
</li>
`;
